ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/lexer/LexerDirectoryProcessor.java

```diff
@@ -0,0 +1,36 @@
+package edu.usb.argos.ASTProcessor.lexer;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class LexerDirectoryProcessor {
+    private final LexerFileProcessor fileProcessor;
+
+    public LexerDirectoryProcessor() {
+        this.fileProcessor = new LexerFileProcessor();
+    }
+
+    public Map<String, CommonTokenStream> getTokensFromDirectoryByFile(String directoryPath) throws IOException {
+        Map<String, CommonTokenStream> tokensByFile = new HashMap<>();
+
+        try (Stream<Path> paths = Files.walk(Paths.get(directoryPath))) {
+            paths.filter(Files::isRegularFile)
+                    .filter(path -> path.toString().endsWith(".java"))
+                    .forEach(path -> {
```

Comment:
It would be good if you refactored this method since you have a forEach and then you do more operations, divide it into small methods so that it is not so complicated to understand.

Reply:
Do you want a method only for a forech?

The method is already simple and straightforward
The code of the getTokensFromDirectoryByFile method has a clear and well-defined purpose:
- Indexer the files in a directory.
- Filter out those that are regular files with .java extension.
- Process each file to get the tokens.

The logic inside the forEach is not complicated, the operation inside the forEach is straightforward:
- Process a file with an already encapsulated method (getTokensFromFile).
- Store the results in a Map.

Reply:
Done ✅